### PR TITLE
remove isGrpcRequest check from the java api to align with scala api

### PR DESCRIPTION
**Pull request by sebastian-alfers**
Fixes https://github.com/akka/akka-grpc/issues/1668 and https://github.com/akka/akka-grpc/issues/1574

### CODE REVIEW

**Review by johanandren — APPROVED** [review]
LGTM

**Comment by johanandren** [comment]
(but let's wait to see what @raboof thinks before merging)

File: runtime/src/main/scala/akka/grpc/javadsl/ServiceHandler.scala [hunk 1]
```diff
@@ -50,7 +49,7 @@ object ServiceHandler {
   def handler(handlers: JFunction[HttpRequest, CompletionStage[HttpResponse]]*)
       : JFunction[HttpRequest, CompletionStage[HttpResponse]] = {
     val servicesHandler = concat(handlers: _*)
-    (req: HttpRequest) => if (sServiceHandler.isGrpcRequest(req)) servicesHandler(req) else unsupportedMediaType
+    (req: HttpRequest) => servicesHandler(req)
```
Comment (raboof) [inline runtime/src/main/scala/akka/grpc/javadsl/ServiceHandler.scala:52]: The comment says "If the request is for an invalid media type, then a _415: Unsupported Media Type_ response is produced.". This probably only is true for paths matched by the handler? Might be something to double-check and perhaps make the comment more specific?

**Review by raboof — APPROVED** [review]
Indeed looks more consistent AFAICS. Would be nice to have more test coverage of the expected behavior in this area, to make sure this 'stays fixed', but if you have tested this for the scenario you care about I'm OK with merging as-is too.

File: runtime/src/main/scala/akka/grpc/javadsl/ServiceHandler.scala (hunk 1 above)
Comment (sebastian-alfers) [reply runtime/src/main/scala/akka/grpc/javadsl/ServiceHandler.scala:50]: Good point, I changed the commend.

**Pull request merged**